src/components/ActivityCollector/createLinkClick.js

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+  getAbsoluteUrlFromAnchorElement,
+  isSupportedAnchorElement,
+  isDownloadLink,
+  isExitLink
+} from "./utils";
+
+const determineLinkType = (window, config, linkUrl, clickedObj) => {
+  let linkType = "other";
+  if (isDownloadLink(config.downloadLinkQualifier, linkUrl, clickedObj)) {
+    linkType = "download";
+  } else if (isExitLink(window, linkUrl)) {
+    linkType = "exit";
+  }
+  return linkType;
+};
+
+export default (window, config, clickedObject) => {
+  let linkName;
+  let linkType;
+  let clickedObj = clickedObject;
+  let linkUrl = getAbsoluteUrlFromAnchorElement(window, clickedObj);
+  let isValidLink = false;
+  // Search parent elements for an anchor element
+  // TODO: Replace with generic DOM tool that can fetch configured properties
+  while (clickedObj && clickedObj !== document.body && !linkUrl) {
+    clickedObj = clickedObj.parentElement || clickedObj.parentNode;
+    if (clickedObj) {
+      linkUrl = getAbsoluteUrlFromAnchorElement(window, clickedObj);
+    }
+  }
+  if (linkUrl && isSupportedAnchorElement(clickedObj)) {
+    isValidLink = true;
+    linkType = determineLinkType(window, config, linkUrl, clickedObj);
+    // TODO: Update link name from the clicked element context
+    linkName = "Link Click";
+  }
+  return {
+    isValid() {
+      return isValidLink;
+    },
+    populateEvent(event) {
+      if (linkType === "exit") {
+        event.documentUnloading();
+      }
+      event.mergeXdm({
+        eventType: "web.webinteraction.linkClicks",
```

Comment:
What happens if the ActivityCollector determines the click is not a valid link, and the Personalization component determines that the clicked on object has attached data that needs to be sent?  How would the personalization component know to set an eventType?  There are a few options to solve for this:

1. Leave things how they are in this PR.  Require the Personalization component to set `eventType: "web.webinteraction.linkClicks"` when adding event data.  This way the order of the lifecycle resolution would not matter.
2. Use multiple events - pass a payload object to onClick and have the lifecycle method implementations add events as needed.
3. Set the event type in the clickHandler.

Reply:
Yeah, this is interesting. Does Personalization only intend to send data for link clicks? Or could it potentially send data for non-link clicks as well? If it's the latter, then `eventType` probably shouldn't be set to `web.webinteraction.linkClicks` in that case, right?

Reply:
We touched on this earlier. The eventType doesn't carry much valuable information when the data it directly correlates to is part of the payload.

Reply:
Sorry, I don't follow. Sorry if I missed the discussion.

Reply:
I will remove eventType for linkClicks for now. If there is a need to put it back we'll have to decide how to change the request.